darkmoon/api/v1/metadata/views.py

```diff
@@ -0,0 +1,205 @@
+"""This is the router file."""
+
+###########
+# IMPORTS #
+###########
+
+from typing import Optional
+
+import bson
+from bson.objectid import ObjectId
+from fastapi import APIRouter, Depends, HTTPException
+from motor.motor_asyncio import AsyncIOMotorCollection
+from pymongo import errors
+
+from darkmoon.api.v1.metadata.schema import Metadata, MetadataEntity
+from darkmoon.core.database import get_file_metadata_collection
+
+####################
+# GLOBAL VARIABLES #
+####################
+
+router = APIRouter(prefix="/metadata", tags=["metadata"])
+
+###########
+# CLASSES #
+
+
+#############
+# FUNCTIONS #
+#############
+
+
+@router.get("/")
+async def list_metadata(
+    collection: AsyncIOMotorCollection = Depends(get_file_metadata_collection),
+    file_name: Optional[str] = None,
+    hash_type: Optional[str] = None,
+    hash: Optional[str] = None,
+) -> list[MetadataEntity]:
```

Comment:
This API signature is incorrect. Since you are accepting 3 optional types, it is still valid for the user to supply nothing at all, or all 3, and neither are edge cases your program considers valid. The more appropriate solution is to allow the user to supply only 1 of a union of the 3 values.

Reply:
#38